cmake/CheckPatch.cmake

```diff
@@ -0,0 +1,29 @@
+find_program(CHECKPATCH checkpatch.pl
+             HINTS ${PROJECT_SOURCE_DIR}/checkpatch)
+find_program(CODESPELL codespell)
+
+add_custom_target(checkpatch)
+
+set(BASE_GIT_REF "origin/master")
```

Comment:
Minor: it's better to fetch the `HEAD` branch from `git remote show origin` output (you can find the example [here](https://git.imun.cloud/dotfiles.git/tree/bin/applypatch.pl#n77), since the default target branch can be changed (remember master to main bullshit, or renaming of Tarantool long-term branches), so it's better when the default is not hardcoded. If you consider this an overkill for this patch, feel free to ignore.

Reply:
Fixed:

```diff
--- a/cmake/CheckPatch.cmake
+++ b/cmake/CheckPatch.cmake
@@ -4,17 +4,28 @@ find_program(CODESPELL codespell)
 
 add_custom_target(checkpatch)
 
-set(BASE_GIT_REF "origin/master")
-if (DEFINED ENV{CHECKPATCH_GITREF})
+execute_process(
+    COMMAND ${GIT} rev-parse --abbrev-ref origin
+    RESULT_VARIABLE GIT_RESULT
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    OUTPUT_VARIABLE BASE_GIT_REF
+)
+
+message(STATUS ${GIT_RESULT} ${BASE_GIT_REF})
+if((NOT ${GIT_RESULT} EQUAL 0) OR (${BASE_GIT_REF} STREQUAL ""))
+    message(FATAL_ERROR "Base Git reference is not found.")
+endif()
+
+if(DEFINED ENV{CHECKPATCH_GITREF})
     set(BASE_GIT_REF "$ENV{CHECKPATCH_GITREF}")
-endif ()
+endif()
 
```

Reply:
Reverted applied patch above because logic in cmake module became complicated. Moreover, one can pass another `BASE_GIT_REF` via env variable .

Reply:
OK then, let's leave it as is.